extensions/wikia/Wall/index/CommentsIndex.class.php

```diff
@@ -0,0 +1,255 @@
+		$this->objectCache[$commentsIndexEntry->getCommentId()] = $commentsIndexEntry;
+
+		$status = $db->replace(
+			'comments_index',
+			null,
+			$commentsIndexEntry->getDatabaseRepresentation(),
+			__METHOD__
+		);
+
+		if ( !$status ) {
+			$this->error( 'Failed to insert comments index entry', $commentsIndexEntry->getDatabaseRepresentation() );
+			return false;
+		}
+
+		// update last_child_comment_id for parent thread
+		if ( !empty( $commentsIndexEntry->getParentCommentId() ) ) {
+			$parentEntry = static::entryFromId( $commentsIndexEntry->getParentCommentId() );
+			$parentEntry->setLastChildCommentId( $commentsIndexEntry->getCommentId() );
+			$this->updateEntry( $parentEntry );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Return a single Comments Index Entry corresponding to the row in comments_index table with matching comment_id
+	 *
+	 * @param int $commentId
+	 * @return CommentsIndexEntry|null
+	 */
+	public function entryFromId( int $commentId ) {
+		if ( isset( $this->objectCache[$commentId] ) ) {
+			return $this->objectCache[$commentId];
+		}
+
+		$dbr = wfGetDB( DB_SLAVE );
+		$row = $dbr->selectRow( 'comments_index', '*', [ 'comment_id' => $commentId ], __METHOD__ );
+
+		// TODO: Monitor if this actually gets executed and remove if not
+		if ( !$row ) {
+			$this->error( 'SUS-1680 - No match for comment id in slave comments_index - retry from master', [
+				'commentId' => $commentId
+			] );
+
+			$dbw = wfGetDB( DB_MASTER );
+			$row = $dbw->selectRow( 'comments_index', '*', [ 'comment_id' => $commentId ], __METHOD__ );
+
+			if ( !$row ) {
+				$this->error( 'SUS-1680 - No match for comment id in master comments_index', [
+					'commentId' => $commentId
+				] );
+				return null;
+			}
+		}
+
+		$entry = CommentsIndexEntry::newFromRow( $row );
+		$this->objectCache[$commentId] = $entry;
+
+		return $entry;
+	}
+
+	/**
+	 * get CommentsIndexEntry objects for a set of ids in a single database query
+	 *
+	 * @see SUS-262
+	 *
+	 * @param int[] $commentIds
+	 * @return CommentsIndexEntry[]
+	 */
+	public function entriesFromIds( array $commentIds ) {
+
+		// a shortcut that prevents "DatabaseBase::makeList: empty input" exception
+		if ( count( $commentIds ) == 0 ) {
+			return [];
+		}
+
+		$dbr = wfGetDB( DB_SLAVE );
+
+		$res = $dbr->select(
+			'comments_index',
+			'*',
+			[ 'comment_id' => $commentIds ],
+			__METHOD__
+		);
+
+		$comments = [];
+
+		while ( $row = $res->fetchObject() ) {
+			$comments[] = CommentsIndexEntry::newFromRow( $row );
+		}
+
+		return $comments;
+	}
+
+	/**
+	 * Move a given thread to a different Message Wall or Forum Board
+	 *
+	 * @param int $threadId page id of thread to move
+	 * @param int $targetPageId page id of destination page
+	 */
+	public function moveThread( int $threadId, int $targetPageId ) {
+		$db = wfGetDB( DB_MASTER );
+
+		$db->update(
+			'comments_index',
+			[ 'parent_page_id' => $targetPageId ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "parent_page_id" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)